float32 encoder1
float32 encoder2
int32 port_num
